### ISSUE TITLE
PR 942 introduces RuntimeError: wrapped C/C++ object of type GraphicsLayoutWidget has been deleted

### ISSUE THREAD

**Issue by titusjan** [pyqtgraph/pyqtgraph#1021]
Since commit 08c0de7, which merges PR #942, I get the following error when I close my application window.

`RuntimeError: wrapped C/C++ object of type GraphicsLayoutWidget has been deleted`

The following program reproduces the issue.
```

import sys
import pyqtgraph as pg
from pyqtgraph.Qt import QtWidgets
from PyQt5.Qt import PYQT_VERSION_STR


class DemoWindow(QtWidgets.QMainWindow):

    def __init__(self, parent=None):
        super().__init__(parent=parent)

        self.mainWidget = QtWidgets.QWidget()
        self.setCentralWidget(self.mainWidget)

        self.mainLayout = QtWidgets.QVBoxLayout()
        self.mainWidget.setLayout(self.mainLayout)

        self.plotItem = pg.PlotItem()
        self.plotItem.plot([3.5, 5, -4, 0, 0, 3])

        self.graphicsLayoutWidget = pg.GraphicsLayoutWidget()
        self.graphicsLayoutWidget.addItem(self.plotItem, 0, 0)

        self.mainLayout.addWidget(self.graphicsLayoutWidget)


def main():
    print(sys.version)
    print("PyQt version: {}".format(PYQT_VERSION_STR))
    print("PyQtGraph version: {}".format(pg.__version__))
    app = QtWidgets.QApplication([])
    win = DemoWindow()
    win.setGeometry(400, 100, 700, 600)
    win.show()
    app.exec()


if __name__ == '__main__':
    main()
```

I tested with PyQt 5.9 on OS-X 10.14.6.

```
3.6.7 | packaged by conda-forge | (default, Feb 28 2019, 02:16:08)  [GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]
PyQt version: 5.9.2
PyQtGraph version: 0.11.0.dev0
```

and also with PyQt 5.6

```
3.6.8 |Anaconda, Inc.| (default, Dec 29 2018, 19:04:46) [GCC 4.2.1 Compatible Clang 4.0.1 tags/RELEASE_401/final)]
PyQt version: 5.6
PyQtGraph version: 0.11.0.dev0
```

In my opinion this PR will cause more issues than it solves. Messing with the python garbage collection will introduce a lot of unexpected issues like this one, while it doesn't solve a real bug, it only helps novice PyQt programmers.

Keeping references to your objects is one of the first things you learn as a PyQt programmer. Encountering this issue is so common that it is almost a rite of passage. There are plenty of StackOverflow posts that address it, just [googling it](https://www.google.com/search?client=firefox-b-d&q=pyqt+window+closes+immediately) already gives three links at the top. 

My personnel recommendation would be to be to roll back this PR and solve it in the documentation and the examples.

**Comment by j9ac9k** [comment]
As the maintainer that merged PR #942 I agree with you @titusjan that we should roll this back, and use documentation and examples to highlight the issue that PR was meant to address.  Before I go undoing another person's PR myself, I'm going to want to get more of consensus from other maintainers here.

Tagging @ixjlyons @Dschoni @2xB  @hhslepicka for input.

**Comment by j9ac9k** [comment]
Also unrelated to this issue, but @titusjan I advise you to be careful w/ Qt 5.6, we no longer test for it, and PySide2 that was running with Qt 5.6 was failing our test suite left and right.

**Comment by j9ac9k** [comment]
I'm going to merge #1022 but I want to leave this issue open for the general discussion on what we should do on the long run.  I'm also going to quote a relevant bit from the PR that should be posted.

From #1022 by @2xB 
> An associated question is whether #942 should be reverted completely. Regarding this: PyQtGraph advertises "learning by doing" and "learning by trying and editing examples", as opposed to a tutorial-rich documentation, which makes it unlikely for people to read through documentation to this topic, especially since they would not know where exactly to search for. But I also understand other opinions here - I would also be fine with it being removed again, if that is favored. This PR gives only the opportunity to keep this warning if wanted.

While I'm leaning toward removing entirely (due to that's how general Qt behavior is) I'm not that strongly opinionated on the matter and would follow whatever consensus emerges.

**Comment by j9ac9k** [comment]
Given we haven't had an update on this issue in a few months, and it seems like the fix ( #1042 ) has addressed the issue at hand, I'm going to close this issue out.
